libraries/chain/hotstuff/hotstuff.cpp

```diff
@@ -149,7 +149,8 @@ valid_quorum_certificate pending_quorum_certificate::to_valid_quorum_certificate
    } else if (is_quorum_met_no_lock()) {
       valid_qc._strong_votes = _strong_votes._bitset;
       valid_qc._weak_votes   = _weak_votes._bitset;
-      valid_qc._sig          = fc::crypto::blslib::aggregate(std::array{_strong_votes._sig, _weak_votes._sig});
+      valid_qc._sig          = _strong_votes._sig;
+      valid_qc._sig.aggregate(_weak_votes._sig);
```

Comment:
Not sure I understand. This block is `is_quorum_met()` which means `s == state_t::weak_achieved || s == state_t::weak_final` because `strong` is the `if` of this `else if`. So there has to be week signature, right?

Reply:
Oh I missed the `else if`.